README.md

```diff
@@ -78,37 +79,61 @@ You might not have to sort the output, if you run through reads
 
 Implement the two algorithms in two tools, `naive` and `lin`, that must be present at the root of the repository once they are built. The test setup checks that they give the correct output on selected data, but you should still carefully test them.
 
-Once you have implemented the tools, fill out the report below. 
+Once you have implemented the tools, fill out the report below.
 
 ## Report
 
-### Insights you may have had while implementing and comparing the algorithms. 
+### Insights you may have had while implementing and comparing the algorithms.
 
-*Describe this here.*
+Implementing the naïve algorithm was straightforward, and so should have been the linear one, for which we chose KMP, but during the initial coding phase we confused `i` and `j` in some places, which caused a few headscratching moments.
 
-### Problems encountered if any. 
+### Problems encountered if any.
 
-*Describe this here.*
+Our own testing revealed a flaw in constructing border arrays, meaning that in some cases, when two matches overlapped in the string by a single character, we would not recognize the second match. To add insult to injury, this was already the second, improved version of calculating the border arrays, after a way too simple approach on the first try.
 
 ### Experiments that verifies the correctness of your implementations.
 
-*Describe this here.*
+In order to verify that both the naïve and the linear algorithm work as intended, we decided to generate a test with the solutions that tries both regular and extreme situations. The test tries the following cases:
 
-### Experiments validating the running time.
+- Short patterns and genomes (chains), with length going from 0 to 10 characters (selected randomly)
+- Random chains
+- Random chains, having a chance of repeating the previous character
+- Random chains, having a chance of having a long sequence made up by a single character
+- Random chains, where a character cannot appear consecutively two or more times.
+- Fibonacci chains, made up by two different characters being concatenated following a Fibonacci distribution
+
+All of these genomes (except the short ones) have a length between 2.000 and 10.000 characters, whilst the patters length varies between 3 and 200. Moreover, for each pair of chains the genome is adapted, forcing to include the pattern a random number of times between 0 and 10 (if the randomness have made up more). The inclusion of this patterns has a high probability of having four extreme situations:
```

Comment:
I know we are all used to writing continental European numbers, but when writing English, the dot, `.`, is a decimal point :)  You want `2,000` and `10,000` here.

That's just a point on the language, though; there is absolutely nothing wrong with your work.